cdap-storage-spi/src/main/java/io/cdap/cdap/spi/data/StructuredTable.java

```diff
@@ -152,6 +152,41 @@ default CloseableIterator<StructuredRow> scan(Range keyRange, int limit, SortOrd
    */
   CloseableIterator<StructuredRow> scan(Field<?> index) throws InvalidFieldException, IOException;
 
+  /**
+   * Read a set of rows from the table matching the index and the key range.
+   * The rows returned will be sorted on the primary key order.
+   *
+   * @param keyRange key range for the scan
+   * @param limit maximum number of rows to return
+   * @param filterIndex the index to filter upon
+   * @return a {@link CloseableIterator} of rows
+   * @throws InvalidFieldException if the field is not part of the table schema, or is not an indexed column,
+   *                               or the type does not match the schema
+   * @throws IOException if there is an error scanning the table
+   */
+  default CloseableIterator<StructuredRow> scan(Range keyRange, int limit, Field<?> filterIndex)
+    throws InvalidFieldException,
+    IOException {
+      return scan(keyRange, limit);
```

Comment:
can we change the default implementation to `throw new UnsupportedOperationException();`? The current default implementation does not apply with the javadoc, and people might get wrong results back for improper implementation of the SPI

Reply:
yes changed it